### PR TITLE
MultiBatchEdition.sol

### PR DESCRIPTION
**Pull request by 0xkarmacoma**
- passes the Edition base spec (i.e. all the usual goodies: EIP2981 royalties, operator filtering, metadata, looks good on OpenSea out of the box, open editions, time limits, supply limits...)
- unlocks >1k editions
- has the same interface as regular Edition as well as SingleBatchEdition

### CODE REVIEW

File: contracts/MultiBatchEdition.sol [hunk 1]
```diff
@@ -0,0 +1,168 @@
...
+
+        return _tokenId;
+    }
+
+    /*//////////////////////////////////////////////////////////////
+                           SS2ERC721 GOODIES
+    //////////////////////////////////////////////////////////////*/
+
+    /// Returns the SSTORE2 pointer for this edition if minted, or 0 if not minted
+    function getPrimaryOwnersPointer(uint256 index) public view override returns (address) {
```
Comment (jorgeavaldez) [inline contracts/MultiBatchEdition.sol:133]: index of the batch right?

**Review by jorgeavaldez — APPROVED** [review]
:fire: :fire: :fire: 

i did have a separate question about the theoretical max supply but you tested for 1 million nfts and that answered my concerns hahahaha

**Comment by 0xkarmacoma** [comment]
> i did have a separate question about the theoretical max supply but you tested for 1 million nfts and that answered my concerns hahahaha

that's right, with open editions there isn't really a theoretical max supply. As long as you keep submitting _full_ and _sorted_ batches, you can keep them coming

File: contracts/MultiBatchEdition.sol (hunk 1 above)
Comment (0xkarmacoma) [reply contracts/MultiBatchEdition.sol:133]: yes, `getPrimaryOwnersPointer(0)` is the pointer for the first batch, `getPrimaryOwnersPointer(1)` second batch and so on

**Pull request merged**